gdal/frmts/wcs/wcsutils.cpp

```diff
@@ -46,45 +47,15 @@ void Swap(double &a, double &b)
 
 int CompareNumbers(const std::string &a, const std::string &b)
 {
-    size_t a_dot = a.find(".");
-    size_t b_dot = b.find(".");
-    std::string a_p = a.substr(0, a_dot);
-    std::string b_p = b.substr(0, b_dot);
-    int d = (int)(a_p.length()) - (int)(b_p.length());
-    if (d < 0) {
-        for (int i = 0; i < -1*d; ++i) {
-            a_p = "0" + a_p;
-        }
-    } else if (d > 0) {
-        for (int i = 0; i < d; ++i) {
-            b_p = "0" + b_p;
-        }
-    }
-    int c = a_p.compare(b_p);
-    if (c < 0) {
-        return -1;
-    } else if (c > 0) {
-        return 1;
-    }
-    a_p = a_dot == std::string::npos ? a : a.substr(a_dot + 1);
-    b_p = b_dot == std::string::npos ? b : b.substr(b_dot + 1);
-    d = (int)(a_p.length()) - (int)(b_p.length());
-    if (d < 0) {
-        for (int i = 0; i < -1*d; ++i) {
-            a_p = a_p + "0";
-        }
-    } else if (d > 0) {
-        for (int i = 0; i < d; ++i) {
-            b_p = b_p + "0";
-        }
-    }
-    c = a_p.compare(b_p);
-    if (c < 0) {
+    double ad = std::stod(a);
```

Comment:
if a or b might not be representable as double (which might be the case when looking at the callers where we trust the value of some XML element), std::stod() can throw an exception. Perhaps use CPLAtof(a.c_str()) instead.
Actually this functions seems overkill. It would be best replaced by more straighforward comparisons in the 4 call places where it is used (which would save double to string conversions)